sys/include/net/ipv6/ext.h

```diff
@@ -31,6 +33,19 @@ extern "C" {
 #define IPV6_EXT_LEN_UNIT   (8U)    /**< Unit in byte for the extension header's
                                      *   length field */
 
+/** @brief Message type to reply IPv6 that a next header has been processed */
+#define GNRC_IPV6_EXT_HANDLE_NEXT_HDR   (0x0399)
+
+/**
+ * @brief   Message content container to process extension headers sequentially.
+ */
+typedef struct {
+    gnrc_netif_t    *netif;     /**< The interface we received the packet */
+    gnrc_pktsnip_t  *current;   /**< The snip of the IP6 header */
+    gnrc_pktsnip_t  *next_hdr;  /**< The next header to be processed */
+    uint8_t         nh_type;    /**< The type of the next header */
+} gnrc_ipv6_ext_hdr_handle_t;
```

Comment:
I think this however by its very name should go into GNRC not `net/ipv6/ext.h` (which is outside the GNRC realm).

Reply:
(not just the name, also the fact that you are using GNRC-specific types here)

Reply:
Don't know, its purpose is only to enable the sequential iteration of extension headers for IPv6 packets.
Although the structure uses gnrc-types I would expect to find such helper in this extension specific headerfile.
Probably I can just remove the prefix `gnrc_` from the typename to avoid confusion.

Reply:
@BytesGalore before you do anything, have a look at my refactoring work first of the past few weeks. It also should very much simplify the handling of the hop-by-hop-header, as it is now treated separately (which we can do, because they always MUST come first according to RFC8200; see https://github.com/RIOT-OS/RIOT/pull/10244). You should now be able to just hook into the according switch-branch with `gnrc_ipv6_ext_demux()` for that handling.